client/src/services/notificationCenter.ts

```diff
@@ -10,84 +14,79 @@ export enum NotificationLevel {
   Error = 'ERROR',
 }
 
-export interface Notification {
-  id: string,
-  title?: string,
-  message: string,
-  level: NotificationLevel,
-  read: boolean,
-  time: DateTime,
-  data: object,
+export class Notification {
+  id: string;
+  time: DateTime;
+  read: boolean;
+  constructor(public message: string,
+    public level: NotificationLevel,
+    public trace: boolean,
+    public data: object,
+    public title?: string,) {
+      this.id = uuid4()
+      this.time = DateTime.now()
+      this.read = false;
+    }
 }
 
 export class NotificationCenter {
   notifications: Notification[];
 
-  constructor() {
+  constructor(public t: ComposerTranslation) {
     this.notifications = [];
+    this.t = t;
   }
 
-  async showModal(
-    title: string,
-    message: string,
-    level: NotificationLevel,
-    addToList: boolean,
-    trace = false,
-    data = {},
-  ): Promise<void> {
-    const notif: Notification = {
-      id: uuid4(),
-      title: title,
-      message: message,
-      level: level,
-      read: false,
-      time: DateTime.now(),
-      data: data,
-    };
-    if (trace) {
+  async showModal(notif: Notification): Promise<void> {
+    if (notif.trace) {
       this.trace(notif);
     }
-    if (addToList) {
-      this.notifications.push(notif);
+    const modalConfig: MsModalConfig = {
+      theme: this.getModalTheme(notif.level),
+      subtitle: notif.message,
+      closeButtonEnabled: true,
+      confirmButton: {
+        disabled: false,
+        label: this.t('Notification.nextButton'),
+        onClick: () => {
+          this.markAsRead(notif.id);
+          modalController.dismiss();
+        },
+      },
+    };
+    if (notif.title) {
+      modalConfig.title = notif.title;
     }
+    const modal = await modalController.create({
+      component: MsModal,
+      canDismiss: true,
+      cssClass: 'notification-modal',
+      componentProps: modalConfig,
+    });
+    await modal.present();
+    const result = await modal.onWillDismiss();
   }
 
-  async showSnackbar(
-    message: string,
-    level: NotificationLevel,
-    addToList: boolean,
-    trace = false,
-    data = {},
-  ): Promise<void> {
-    const notif: Notification = {
-      id: uuid4(),
-      message: message,
-      level: level,
-      read: false,
-      time: DateTime.now(),
-      data: data,
-    };
-    if (trace) {
+  async showSnackbar(notif: Notification): Promise<void> {
```

Comment:
Keep a `addToList` option. Maybe I want my notification to create a snack and to be in the list of notifications as well. For example, something important like `Hey you joined a new organization!` could be in both. Granted, I could call `showSnackbar()` and then `addToList()` but it means that I cannot create the notification in place.
IE I would have to do
`const notif = new Notification(); showSnackbar(notif); addToList(notif);`
instead of
`showSnackBar(new Notification(), addToList=true);`

Also, there's probably very few cases where you would need to call `addToList` directly, because it would mean not warning the user directly that something as happened, so the prefered way to add a notif to the list should be with `showSnackbar`.

Reply:
Move trace as a parameter of showSnackbar and showModal